src/qml/editorwidgets/relationeditors/relation_editor.qml

```diff
@@ -13,134 +13,147 @@ EditorWidgetBase {
+      } else {
+        maximumVisibleItems = 4
+      }
+    }
 
     // because no additional addEntry item on readOnly (isEnabled false)
-    height: isEnabled
+    height: (isEnabled
             ? referencingFeatureListView.height + itemHeight
-            : Math.max( referencingFeatureListView.height, itemHeight)
+            : Math.max( referencingFeatureListView.height, itemHeight)) + 10
     enabled: true
 
-    Rectangle {
-        anchors.fill: parent
-        color: "transparent"
-        border.color: 'lightgray'
-        border.width: 1
-    }
-
     ReferencingFeatureListModel {
-        //containing the current (parent) feature, the relation to the children
-        //and the relation from the children to the other parent (if it's nm and cardinality is set)
-        //if cardinality is not set, the nmRelationId is empty
-        id: relationEditorModel
-        currentRelationId: relationId
-        currentNmRelationId: nmRelationId
-        feature: currentFeature
-
-        property int featureFocus: -1
-        onModelUpdated: {
-          if (featureFocus > -1) {
-            referencingFeatureListView.currentIndex = relationEditorModel.getFeatureIdRow(featureFocus)
-            featureFocus = -1
-          }
+      //containing the current (parent) feature, the relation to the children
+      //and the relation from the children to the other parent (if it's nm and cardinality is set)
+      //if cardinality is not set, the nmRelationId is empty
+      id: relationEditorModel
+      currentRelationId: relationId
+      currentNmRelationId: nmRelationId
+      feature: currentFeature
+
+      property int featureFocus: -1
+      onModelUpdated: {
+        if (featureFocus > -1) {
+          referencingFeatureListView.currentIndex = relationEditorModel.getFeatureIdRow(featureFocus)
+          featureFocus = -1
         }
+      }
     }
 
-    //the list
-    ListView {
-        id: referencingFeatureListView
-        model: relationEditorModel
+    Rectangle {
+        anchors.top: parent.top
+        anchors.bottom: parent.bottom
+        anchors.bottomMargin: 10
         width: parent.width
-        height: Math.min( 4 * itemHeight, referencingFeatureListView.count * itemHeight ) + ( referencingFeatureListView.count > 4 ? itemHeight / 2 : 0 )
-        delegate: referencingFeatureDelegate
-        focus: true
+        color: "transparent"
+        border.color: 'lightgray'
+        border.width: 1
         clip: true
-        highlightRangeMode: ListView.StrictlyEnforceRange
 
-        ScrollBar.vertical: ScrollBar {
+        //the list
+        ListView {
+          id: referencingFeatureListView
+          model: relationEditorModel
+          width: parent.width
+          height: Math.min( maximumVisibleItems * itemHeight, referencingFeatureListView.count * itemHeight ) + ( referencingFeatureListView.count > maximumVisibleItems ? itemHeight / 2 : 0 )
```

Comment:
```suggestion
          height: Math.min(
            maximumVisibleItems * itemHeight,
            referencingFeatureListView.count * itemHeight
          ) + (
            referencingFeatureListView.count > maximumVisibleItems ? itemHeight / 2 : 0
          )
```

Sorry, it's super hard to read these long lines.

Reply:
So speaking on this line, would be nice to have an option that follows the number of items, no matter what. I already discussed this here:
https://github.com/opengisch/qfieldsync/pull/474#discussion_r1006849612

Since you assuume the number 4 by default, probably should be a reserved value, e.g. -1, or 0.